realm/src/main/java/io/realm/Realm.java

```diff
@@ -60,33 +60,59 @@
 
 
 /**
- * <p>The Realm class is the storage and transactional manager of your object persistent store. It is in charge of
- * creating instances of your RealmObjects.
- * Objects within a Realm can be queried and read at any time. Creating,
- * modifying, and deleting objects must be done through transactions.</p>
+ * The Realm class is the storage and transactional manager of your object persistent store. It
+ * is in charge of creating instances of your RealmObjects. Objects within a Realm can be queried
+ * and read at any time. Creating, modifying, and deleting objects must be done while inside a
+ * transaction. See {@link #beginTransaction()}
  *
- * <p>The transactions ensure that multiple instances (on multiple threads) can access the objects
- * in a consistent state with full ACID guaranties.</p>
+ * The transactions ensure that multiple instances (on multiple threads) can access the same
+ * objects in a consistent state with full ACID guarantees.
  *
- * <p>If auto-refresh is set the instance of the Realm will be automatically updated when another instance commits a
- * change (create, modify or delete an object). This feature requires the Realm instance to be residing in a
- * thread attached to a Looper (the main thread has a Looper by default)</p>
+ * It is important to remember to call the {@link #close()} method when done with a Realm
+ * instance. Failing to do so can lead to {@link java.lang.OutOfMemoryError} as the native
+ * resources cannot be freed.
  *
- *<p>For normal threads Android provides a utility class that can be used like this:</p>
+ * Realm instances cannot be used across different threads. This means that you have to open an
+ * instance on each thread you want to use Realm. Realm instances are cached automatically per
+ * thread using reference counting, so as long as the reference count doesn't reach zero, calling
+ * {@link #getInstance(android.content.Context)} will just return the cached Realm and should be
+ * considered a lightweight operation.
+ *
+ * For the UI thread this means that opening and closing Realms should occur in either
+ * onCreate/onDestroy or onStart/onStop. onResume/onPause do not overlap between new activities,
+ * which mean the reference count will reach zero and the Realm instance will be fully closed,
```

Comment:
This sentence is hard to comprehend. Also s/mean/means that/ if you decide to keep it as it is.